### ISSUE TITLE
should  json-api differentiate readAs/actAs for read/write endpoints?

### ISSUE THREAD

**Issue by lima-da** [digital-asset/daml#4630]
json-api currently does not differentiate readAs/actAs in jwt before sending requests to sandbox. 
it only decode a party field before calling ledgerClient. when query-store is enabled, it's non-deterministic a token is really having read access when reading from cache. This behavior needs to be discussed for either a patch or documentation.

**Comment by cocreature** [comment]
There are 2 cases to consider here, note for everything here that `actAs` implies `readAs`:

1. Write endpoints: Those always go to the ledger which validates the token.
2. Read endpoints: Some of them hit the ledger so they are validated there. The interesting point is when we do not query the ledger for contracts. This only happens if the ledger offset hasn’t changed. However, to get the current ledger offset, we query the ledger so the ledger validates that the token is signed correctly. It does not validate that the token has the correct party in `readAs` or `actAs` but the JSON API takes care of that.

So in summary, there is no issue here but arguably the ledger offset part is somewhat by accident rather than deliberate design. I’m closing this for now.
